python/ray/train/v2/_internal/state/export.py

```diff
@@ -0,0 +1,106 @@
+from ray.core.generated.export_train_state_pb2 import (
+    ExportTrainRun as ProtoTrainRun,
+    ExportTrainRunAttempt as ProtoTrainRunAttempt,
+)
+from ray.train.v2._internal.state.schema import (
+    TrainRunAttempt,
+    TrainRun,
+    TrainWorker,
+    RunAttemptStatus,
+    RunStatus,
+    ActorStatus,
+)
+
+
+# Status mapping dictionaries
+_ACTOR_STATUS_MAP = {
+    ActorStatus.ALIVE: ProtoTrainRunAttempt.ActorStatus.ALIVE,
+    ActorStatus.DEAD: ProtoTrainRunAttempt.ActorStatus.DEAD,
+}
+
+_RUN_ATTEMPT_STATUS_MAP = {
+    RunAttemptStatus.PENDING: ProtoTrainRunAttempt.RunAttemptStatus.PENDING,
+    RunAttemptStatus.RUNNING: ProtoTrainRunAttempt.RunAttemptStatus.RUNNING,
+    RunAttemptStatus.FINISHED: ProtoTrainRunAttempt.RunAttemptStatus.FINISHED,
+    RunAttemptStatus.ERRORED: ProtoTrainRunAttempt.RunAttemptStatus.ERRORED,
+    RunAttemptStatus.ABORTED: ProtoTrainRunAttempt.RunAttemptStatus.ABORTED,
+}
+
+_RUN_STATUS_MAP = {
+    RunStatus.INITIALIZING: ProtoTrainRun.RunStatus.INITIALIZING,
+    RunStatus.SCHEDULING: ProtoTrainRun.RunStatus.SCHEDULING,
+    RunStatus.RUNNING: ProtoTrainRun.RunStatus.RUNNING,
+    RunStatus.RESTARTING: ProtoTrainRun.RunStatus.RESTARTING,
+    RunStatus.RESIZING: ProtoTrainRun.RunStatus.RESIZING,
+    RunStatus.FINISHED: ProtoTrainRun.RunStatus.FINISHED,
+    RunStatus.ERRORED: ProtoTrainRun.RunStatus.ERRORED,
+    RunStatus.ABORTED: ProtoTrainRun.RunStatus.ABORTED,
+}
+
+
+# Helper conversion functions
+def _ms_to_ns(time_ms: int) -> int:
+    """Convert millisecond timestamp to ns."""
+    return time_ms * int(1e6)
+
+
+def _to_proto_resources(resources: dict) -> ProtoTrainRunAttempt.TrainResources:
+    """Convert resources dictionary to protobuf TrainResources."""
+    return ProtoTrainRunAttempt.TrainResources(resources=resources)
+
+
+def _to_proto_worker(worker: TrainWorker) -> ProtoTrainRunAttempt.TrainWorker:
+    """Convert TrainWorker to protobuf format."""
+    status = None
+    if worker.status is not None:
+        status = _ACTOR_STATUS_MAP[worker.status]
+
+    return ProtoTrainRunAttempt.TrainWorker(
+        world_rank=worker.world_rank,
+        local_rank=worker.local_rank,
+        node_rank=worker.node_rank,
+        actor_id=bytes.from_hex(worker.actor_id),
+        node_id=bytes.from_hex(worker.node_id),
```

Comment:
I convert to bytes at the end when creating the export data rather than keeping it as bytes in the internal train state schema, since the public `ray.get_runtime_context().get_job_id()` returns it as a string, and also since the string is easier to work with if we want to log it internally.